tests/test_universe.py

```diff
@@ -0,0 +1,413 @@
+# Copyright (c) 2023, Yung-Yu Chen <yyc@solvcon.net>
```

Comment:
The test file is rather long, but a good starting point to understand the calculation.